cynic-querygen/src/lib.rs

```diff
@@ -100,16 +100,33 @@ pub fn document_to_fragment_structs(
 
     let mut output = String::new();
 
+    // https://github.com/obmarg/cynic/issues/71
+    let scalar_count = schema
+        .definitions
+        .iter()
+        .filter(|def| {
+            if let schema::Definition::TypeDefinition(schema::TypeDefinition::Scalar(_)) = def {
+                true
+            } else {
+                false
+            }
```

Comment:
You should be able to use `matches!` to make this a bit more concise:

```suggestion
            matches!(def, schema::Definition::TypeDefinition(schema::TypeDefinition::Scalar(_)))
```

Reply:
I'd also consider using the [`Iterator::any`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) function instead of `filter(...).count()` - would give a boolean you could use instead of a count.  Although filter.count is also fine if you'd prefer.